packages/grafana-ui/src/components/Table/TableCellBuilder.tsx

```diff
@@ -291,3 +291,33 @@ class CellBuilderWithStyle {
     return simpleCellBuilder({ value, props, className });
   };
 }
+
+export function getFieldCellBuilder(field: Field, style: ColumnStyle | null, p: Props): TableCellBuilder {
+  if (!field.display) {
+    return getCellBuilder(field.config || {}, style, p);
+  }
+
+  return (cell: TableCellBuilderOptions) => {
+    const { props } = cell;
+    const disp = field.display!(cell.value);
+
+    let style = props.style;
+    if (disp.color) {
+      style = {
+        ...props.style,
+        background: disp.color,
+      };
+    }
+
+    let clazz = 'gf-table-cell';
+    if (cell.className) {
+      clazz += ' ' + cell.className;
```

Comment:
Nit: This could be simplified using Emotion's `cx` function to compose classes. S the final class name would simply be `cx('gf-table-cell', cell.className)`.  No nieed for conditions.